perfect_serving_for_division.py

```diff
@@ -0,0 +1,25 @@
+
+
+def find_perfect_serving():
+    """
+    This is a generator that runs in an endless loop and returns using yield if the sum of all it's
+    divisions equals to the number itself.
+    :return: using yield returns a number which the sum of all of it's divisions equals to the number
+    itself.
+    """
+    # The number we want to check if it's division's sum equals to it.
+    number_of_serving = 1
+    # The sum of all divisions
+    sum_of_divisions = 0
+    while True:
+        # The result variable is just to prevent a warning that the statement has no effect
+        result = [(yield number_of_serving) if sum(sum_of_divisions + serving_division if number_of_serving %
```

Comment:
I know that we encourage you to use list comprehension but in my opinion this line is too long and it is a but hard for reading.
Can you try to split it? (without giving up on the list comprehention structure).